execution/gethexec/executionengine.go

```diff
@@ -358,17 +361,20 @@ func (s *ExecutionEngine) sequenceTransactionsWithBlockMutex(header *arbostypes.
 		return nil, err
 	}
 
+	pos, err := s.BlockNumberToMessageIndex(lastBlockHeader.Number.Uint64() + 1)
+	if err != nil {
+		return nil, err
+	}
+
 	msgWithMeta := arbostypes.MessageWithMetadata{
 		Message:             msg,
 		DelayedMessagesRead: delayedMessagesRead,
 	}
-
-	pos, err := s.BlockNumberToMessageIndex(lastBlockHeader.Number.Uint64() + 1)
-	if err != nil {
-		return nil, err
+	msgResult := execution.MessageResult{
+		BlockHash: block.Hash(),
```

Comment:
you need to also have sendRootHere. Use resultFromHeader

Reply:
Done.